client/js/commands/join.js

```diff
@@ -8,14 +8,37 @@ exports.input = function(args) {
 	const {vueApp} = require("../vue");
 
 	if (args.length > 0) {
-		const channel = args[0];
+		let channel = args[0];
 
 		if (channel.length > 0) {
+			const chanTypes = vueApp.activeChannel.network.serverOptions.CHANTYPES;
+			let hasPrefix = false;
+
+			for (const prefix of chanTypes) {
+				if (channel.startsWith(prefix)) {
+					hasPrefix = true;
+					break;
+				}
+			}
+
+			if (!hasPrefix && chanTypes && chanTypes.length > 0) {
+				channel = chanTypes[0] + channel;
+			}
+
 			const chan = utils.findCurrentNetworkChan(channel);
 
 			if (chan) {
 				$(`#sidebar .chan[data-id="${chan.id}"]`).trigger("click");
 			}
+
+			if (!hasPrefix) {
+				socket.emit("input", {
+					text: `/join ${channel} ${args.length > 1 ? args[1] : ""}`,
```

Comment:
I do wonder if need to care about comma separated channels because you can join multiple channels at once.

/join #chan1,#chan2
/join chan1,#chan2
/join #chan1,chan2

etc. How should we handle prefixes here?

https://tools.ietf.org/html/rfc2812#section-3.2.1

Reply:
Good question. Joining a list of channels does not appear to work without these changes either.

Reply:
It does work for me. (without this pr at least)

Reply:
You're right, it does. I had a space after the comma in the list. I suppose I could split the list up and loop through.

Reply:
There, that should work for all scenarios you posted.